client/packages/programs/src/JsonForms/components/PeriodSearch.tsx

```diff
@@ -63,20 +63,31 @@ const UIComponent = (props: ControlProps) => {
     ? (data.pages[data.pages.length - 1]?.pageNumber ?? 0)
     : 0;
 
-  const onChange = async (period: PeriodFragment) => {
+  const onChange = async (period: PeriodFragment | null) => {
     setPeriod(period);
-
-    if (path === 'periodId') {
-      handleChange(path, period.id);
+    if (period === null) {
+      handleChange(path, undefined);
+      handleChange('before', undefined);
     } else {
-      // date range so we can use it if no period id is saved
-      handleChange(path, new Date(period.startDate).toISOString());
-      const endOfDay = new Date(period.endDate);
-      endOfDay.setHours(24, 59, 59, 999);
-      handleChange('before', endOfDay.toISOString());
+      if (path === 'periodId') {
+        handleChange(path, period.id);
+      } else {
+        // date range so we can use it if no period id is saved
+        handleChange(path, new Date(period.startDate).toISOString());
+        const endOfDay = new Date(period.endDate);
+        endOfDay.setHours(24, 59, 59, 999);
+        handleChange('before', endOfDay.toISOString());
+      }
     }
   };
 
+  useEffect(() => {
+    if (options?.findByProgram) {
+      setPeriod(null);
+      onChange(null);
+    }
+  }, [programId]);
```

Comment:
This is to handle clearing of period filter if the program changes which the period is dependent on.

Now both inputs are cleared when the program is changed.